### ISSUE TITLE
SC2153 diagnoses match with similar read only variable

### ISSUE THREAD

**Issue by mshawcroft** [koalaman/shellcheck#2296]
#### For bugs
- SC2153
- My shellcheck version 0.7.2 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
if [ -n "${GERRIT_HOST:-}" ]; then
  echo "$GERRIT_SCHEME://$GERRIT_HOST${GERRIT_PORT:+:$GERRIT_PORT}"
fi
```

#### Here's what shellcheck currently says:
  echo "$GERRIT_SCHEME://$GERRIT_HOST${GERRIT_PORT:+:$GERRIT_PORT}"
                                                     ^----------^ SC2153: Possible misspelling: GERRIT_PORT may not be assigned, but GERRIT_HOST is.

#### Here's what I wanted or expected to see:

No diagnostic.

The GitHub page https://github.com/koalaman/shellcheck/wiki/SC2153 states "This error only triggers for environment variables (all uppercase variables), and only when they have names similar to something assigned in the script. "

In this example given the diagnostic triggers for an environment variable with a similar name that is not assigned in the script.

**Comment by koalaman** [comment]
You're right, this is confusing for two reasons:

1. The message and docs says the suggested variable is assigned, which used to be true, but has since evolved a bit and should say "known" (as it's been checked with `[ -n .. ]`). I've updated the message and docs.
2. `$GERRIT_PORT` is checked via `:+`, semantically the same as `$GERRIT_HOST` via `[ -n .. ]`, so a warning should be avoided for the same reason.

Arguably a third reason is GERRIT_HOST and GERRIT_PORT seem like completely unrelated variables, but it's hard for ShellCheck to know that two variables matching `GERRIT_?O?T` are so dissimilar to humans.

These issues have now been improved, thanks!
